### PR TITLE
[AIRFLOW-3929] Use anchor tag for dag filter link.

### PR DESCRIPTION
**Pull request by jmcarp**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3929
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Replace a lingering button with an anchor to finish AIRFLOW-3929.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by jmcarp** [comment]
Ready for review when you have time @ashb @mik-laj. This resolves one last span that should be an anchor that I missed in my previous patch.

File: airflow/www/templates/airflow/dag.html [hunk 1]
```diff
@@ -125,12 +125,12 @@ <h4 class="pull-right">
       <div class="modal-content">
         <div class="modal-header">
           <h4 class="modal-title" id="myModalLabel">
-            <span id='task_id'></span>
-            <button id="btn_filter" type="button" class="btn btn-default" title="Filter on this task and upstream ">
+            <span id="task_id"></span>
+            <a id="btn_filter" type="button" class="btn btn-default" title="Filter on this task and upstream ">
```
Comment (ashb) [inline airflow/www/templates/airflow/dag.html]: `<a type=button>` is not a valid property :)

Comment (jmcarp) [reply airflow/www/templates/airflow/dag.html]: Thanks, updated to drop `type="button"` from all anchors in this template.

**Review by ashb — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]

**Comment by jmcarp** [comment]
Tests finally passing 🎉

**Comment by jmcarp** [comment]
Ping @ashb

**Comment by ashb** [comment]
@jmcarp I had to make some changes to these modals in https://github.com/apache/airflow/pull/5039 - making any "mutating" action a form that is POSTed. so this will need fixing up.

Also I noticed that on master, possibly as a result of https://github.com/apache/airflow/blob/c552996483f5aab5e31793317b26aba62c13332e/airflow/www/templates/airflow/dag.html#L336-L339

Before it looked like

https://github.com/apache/airflow/blob/bb91246cfc6f7905c84aea4a44c11aa057b8f42c/airflow/www/templates/airflow/dag.html#L388-L393

1. we can't zoom in to subdags anymore
2. From the graph view the Zoom button _also_ takes you to the tree view, instead of staying on the current view.

**Comment by OmerJog** [comment]
@ashb There is a pending pr to fix the zoom in issue: https://github.com/apache/airflow/pull/4473

**Pull request closed without merging**